continuous-integration/starter-kits/README.md

```diff
@@ -1,6 +1,177 @@
+phases handoff processing between steps. All implementations are predicated on the following 
+configurations:
+1. Setup an account on a Jenkins server to host the build, such as the enterprise CAE Jenkins server 
+at JPL or another shared Jenkins host. 
+2. [Setup Webhook notifications](https://github.blog/2014-02-11-webhooks-level-up/) pointing to the 
+Jenkins instance with the URL of the server from the previous step that will host the build.
+3. Setup an account on an Artifactory server to host built binaries, such as the enterprise CAE 
+Artifactory server at JPL or another similar repository.
+4. Setup authorization credentials to interconnect services.
+   1. Setup [Github SSH Key Access](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/about-ssh) 
+   and record the key for later use.
+   2. Create an [API Key in Artifactory](https://www.jfrog.com/confluence/display/JFROG/User+Profile#UserProfile-APIKey) 
+   and record it for later use.
+   3. Create a [Maven Settings File](https://www.jfrog.com/confluence/display/JFROG/Maven+Repository) 
+   using the previously generated Artifactory API Key
+5. [Configure Jenkins Global Credentials](https://www.jenkins.io/doc/book/using/using-credentials/#configuring-credentials) 
+by creating an entry for each of the Github SSH key and Maven Settings File previously generated.
+
+#### Base configs and alternatives
+These examples rely on certain assumptions about underlying tools. For instance, a 
+[Dockerfile](https://github.com/nasa-ammos/slim/blob/main/continuous-integration/starter-kits/compile-release-publish/java-maven/simple/.ci/Dockerfile.buildDeploy) 
+referenced in one example presumes that Jenkins uses a certain UID and GID set up for a particular 
+Jenkins server. These values may be changed based on any chosen target server. 
+
+Also, this starter kit should be considered generally as a guided template that is flexible enough to 
+be reprovisioned to work with any number of enterprise or publicly-available, hosted, open-source 
+servers. For instance Github and Github Enterprise may be provisioned similarly for Webhooks, even 
+when connecting to one of several public (or privately hosted) CI build servers. It's a specific example 
+of [general architecture](https://nasa-ammos.github.io/slim/continuous-integration/reference-architectures/).
+
+### Maven `simple` project
+Starter Kit:
+- [Maven Simple Project Root](https://github.com/nasa-ammos/slim/blob/main/continuous-integration/starter-kits/compile-release-publish/java-maven/simple/)
+
+To leverage this example, make sure to do the following: (Of course, using earlier requirements.)
+1. Check out the Git repository to convert into a new folder.
+2. Copy the `simple` project layout onto the folder from item (1).  
+3. Move source and configuration files into place per the [Maven Standard Directory Layout](https://maven.apache.org/guides/introduction/introduction-to-the-standard-directory-layout.html)
```

Comment:
I think using an archetype pattern would also simplify the customization, i.e. `INSERT XYZ HERE`. There's a lot of repetition of variable insertion, and we could automate that up-front using an archetype (command-line) form.

Reply:
Originally, I didn't think this was something we would want since archetypes are so esoteric. But now that I think more about it this might be a better approach.